manifest/pack_pull_git.go

```diff
@@ -6,18 +6,99 @@ package manifest
 
 import (
 	"context"
-	"encoding/hex"
 	"fmt"
+	"io"
+	"math"
+	"math/rand"
 	"net/url"
 	"strings"
+	"sync"
+	"time"
 
-	libgit "github.com/libgit2/git2go/v31"
+	"github.com/go-git/go-git/v5"
+	gitplumbing "github.com/go-git/go-git/v5/plumbing"
+	githttp "github.com/go-git/go-git/v5/plumbing/transport/http"
+	gitssh "github.com/go-git/go-git/v5/plumbing/transport/ssh"
 
 	"kraftkit.sh/log"
 	"kraftkit.sh/pack"
 	"kraftkit.sh/unikraft"
 )
 
+type cloneProgress struct {
+	io.Writer
+
+	once           sync.Once
+	onProgress     func(progress float64)
+	completeParent chan bool
+	completeWorker chan bool
```

Comment:
The boolean value isn't used, so we may as well use zero values (an empty struct is usually considered idiomatic for that).

```suggestion
	completeParent chan struct{}
	completeWorker chan struct{}
```